compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1396,13 +1396,28 @@ impl<'context> Elaborator<'context> {
             let trait_constraint_type = trait_constraint.typ.substitute(&bindings);
             let trait_bound = &trait_constraint.trait_bound;
 
+            // Replace implicitly added named generics with fresh type variables
```

Comment:
Should mention here that the impl uses implicitly added named generics to represent its associated types and we need to replace them if they are unbound because they cannot be bound over until monomorphization, so they would not match anything otherwise. If they are bound we do not replace them because we want to ensure the binding matches.

Reply:
Done!